CMakeLists.txt

```diff
@@ -35,25 +35,22 @@ option(MRDOX_BUILD_TESTS "Build tests" ${BUILD_TESTING})
 #
 #-------------------------------------------------
 
-function(get_llvm_dir_for_build_type build_type output_var)
-  if(${build_type} STREQUAL "Debug")
-    set(llvm_dir ${LLVM_DEBUG_PATH})
-  elseif(${build_type} STREQUAL "DebugRel")
-    set(llvm_dir ${LLVM_DEBUGREL_PATH})
-  elseif(${build_type} STREQUAL "RelWithDebInfo")
-    set(llvm_dir ${LLVM_RELWITHDEBINFO_PATH})
-  else()
-    message(FATAL_ERROR "Unsupported build type: ${build_type}")
-  endif()
-
-  set(${output_var} ${llvm_dir} PARENT_SCOPE)
-endfunction()
-
-get_llvm_dir_for_build_type(${CMAKE_BUILD_TYPE} LLVM_DIR)
+set(LLVM_CONFIG_PATH_DEBUG "" CACHE PATH "Path to LLVM Debug installation")
+set(LLVM_CONFIG_PATH_RELEASE "" CACHE PATH "Path to LLVM Release installation")
+set(LLVM_CONFIG_PATH_RELWITHDEBINFO "" CACHE PATH "Path to LLVM RelWithDebInfo installation")
+
+if(NOT LLVM_CONFIG_PATH_DEBUG OR NOT LLVM_CONFIG_PATH_RELEASE OR NOT LLVM_CONFIG_PATH_RELWITHDEBINFO)
+    message(FATAL_ERROR "Please provide paths to LLVM installations for Debug, Release, and RelWithDebInfo configurations")
+endif()
 
 set(CMAKE_FOLDER Dependencies)
 
-find_package(LLVM REQUIRED CONFIG PATHS ${LLVM_DIR})
+find_package(LLVM REQUIRED CONFIG
+    PATHS
+        $<$<CONFIG:Debug>:${LLVM_CONFIG_PATH_DEBUG}>
```

Comment:
Generator expressions won't work here

https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#id1

Either LLVM provides an LLVMConfig that works with multi-config or we are stuck with a single version in multi-config.